In [1]:
import pandas as pd
import numpy as np
import datetime as dt 

In [2]:
df = pd.read_json('a3cleanedonly2015.json')
# df.head(20)

In [3]:
len(df['UID'].unique())

1537

In [4]:
# Number of races 

print(df['Race'].unique())
df['Race'].value_counts()


['Asian' 'Black' 'Hispanic' 'Other' 'Native' 'White' '']


White       751
Black       378
Hispanic    246
            114
Asian        19
Native       19
Other        10
Name: Race, dtype: int64

Data cleaning: Dropped column 'Unamed: 0' and 'FIELD1'. Using UID as the unique identifier. 
We have 1537 data points. 

114 missing points for Race (7% of the data). We converted it to 'Unknown'.

In [5]:
df.drop(columns = ['FIELD1', 'Unnamed: 0'], inplace = True)

In [6]:
df['Race'] = np.where(df['Race'] == '', 'Unknown', df['Race'])
df['Race'].value_counts()

White       751
Black       378
Hispanic    246
Unknown     114
Asian        19
Native       19
Other        10
Name: Race, dtype: int64

In [7]:
df.head(10)

UID                     Name   Age Gender   Race       Date  \
0   9978              Tim Elliott  53.0   Male  Asian 2015-01-02   
1  10077               Hung Trieu   NaN   Male  Asian 2015-02-01   
2  10374                 Lue Vang  39.0   Male  Asian 2015-04-22   
3  10421            Thong Kien Ma  32.0   Male  Asian 2015-05-05   
4  10628               Ton Nguyen  60.0   Male  Asian 2015-07-02   
5  10912             William Chau  59.0   Male  Asian 2015-09-09   
6  10918                  Tian Ma  31.0   Male  Asian 2015-09-10   
7  11009                Kevin Lau  44.0   Male  Asian 2015-10-11   
8  11166               Tuan Hoang   NaN   Male  Asian 2015-11-30   
9  11219  Mharloun Verdejo Saycon  39.0   Male  Asian 2015-12-14   

               City State Manner_of_death  Armed  Mental_illness   Flee  
0           Shelton    WA            Shot                   True  False  
1           Houston    TX            Shot                  False  False  
2  Eldorado Springs    CO            Shot                  False  False  
3    South El Monte    CA            Shot                  False  False  
4         San Diego    CA            Shot                  False  False  
5          El Monte    CA            Shot  Knife           False  False  
6           Potsdam    NY            Shot                  False  False  
7      Lake Charles    LA            Shot    Gun           False  False  
8            Aurora    CO            Shot                  False  False  
9        Long Beach    CA            Shot  Knife           False  False

In [8]:
# Grouping data by month 

# Extract month from Date 

df['Month'] = df['Date'].dt.month
df['Month'] = pd.to_datetime(df['Month'], format='%m').dt.month_name()

# df

In [9]:
# Group by month 
monthly_shootings = df.groupby(['Month', 'Race'])['UID'].count().reset_index()
monthly_shootings.rename(columns = {'UID' : 'Count'}, inplace = True)
monthly_shootings

Month      Race  Count
0       April     Asian      2
1       April     Black     41
2       April  Hispanic     22
3       April    Native      3
4       April   Unknown      7
..        ...       ...    ...
63  September  Hispanic     20
64  September    Native      7
65  September     Other      1
66  September   Unknown     12
67  September     White     59

[68 rows x 3 columns]

In [10]:
month_and_race = monthly_shootings.pivot(index='Month', columns='Race', values='Count')
month_and_race

Race       Asian  Black  Hispanic  Native  Other  Unknown  White
Month                                                           
April        2.0   41.0      22.0     3.0    NaN      7.0   64.0
August       NaN   33.0      23.0     NaN    NaN     15.0   65.0
December     4.0   31.0      23.0     NaN    4.0      9.0   57.0
February     1.0   29.0      19.0     NaN    NaN      9.0   59.0
January      4.0   26.0      21.0     3.0    NaN      6.0   58.0
July         1.0   31.0      25.0     2.0    2.0      8.0   83.0
June         NaN   23.0      24.0     1.0    NaN      6.0   49.0
March        2.0   42.0      18.0     3.0    NaN     11.0   74.0
May          1.0   31.0       8.0     NaN    2.0     17.0   67.0
November     1.0   27.0      20.0     NaN    1.0      NaN   62.0
October      1.0   35.0      23.0     NaN    NaN     14.0   54.0
September    2.0   29.0      20.0     7.0    1.0     12.0   59.0

In [11]:
month_and_race.to_csv('/Users/salma/Documents/GitHub/CAPP30239-FA22/week_06/police_hw/month_and_race.csv')

In [12]:
test = pd.read_csv('/Users/salma/Documents/GitHub/CAPP30239-FA22/week_06/police_hw/month_and_race.csv')
# test

In [13]:
# GRAPH 2:Shootings per month 

by_month = df.groupby('Month')['UID'].count().reset_index()
by_month.rename(columns = {'UID' : 'Count'}, inplace = True)
by_month


Month  Count
0       April    139
1      August    136
2    December    128
3    February    117
4     January    118
5        July    152
6        June    103
7       March    150
8         May    126
9    November    111
10    October    127
11  September    130

In [18]:
by_month.to_csv('/Users/salma/Documents/GitHub/CAPP30239-FA22/week_06/police_hw/monthly_shootings.csv', index = False)


In [17]:
test = pd.read_csv('/Users/salma/Documents/GitHub/CAPP30239-FA22/week_06/police_hw/month_and_race.csv')
test

Month  Asian  Black  Hispanic  Native  Other  Unknown  White
0       April    2.0   41.0      22.0     3.0    NaN      7.0   64.0
1      August    NaN   33.0      23.0     NaN    NaN     15.0   65.0
2    December    4.0   31.0      23.0     NaN    4.0      9.0   57.0
3    February    1.0   29.0      19.0     NaN    NaN      9.0   59.0
4     January    4.0   26.0      21.0     3.0    NaN      6.0   58.0
5        July    1.0   31.0      25.0     2.0    2.0      8.0   83.0
6        June    NaN   23.0      24.0     1.0    NaN      6.0   49.0
7       March    2.0   42.0      18.0     3.0    NaN     11.0   74.0
8         May    1.0   31.0       8.0     NaN    2.0     17.0   67.0
9    November    1.0   27.0      20.0     NaN    1.0      NaN   62.0
10    October    1.0   35.0      23.0     NaN    NaN     14.0   54.0
11  September    2.0   29.0      20.0     7.0    1.0     12.0   59.0